Added all of my mods and Gareth48's

badhaloninja opened the pull request:
Added all of my missing mods and updated ones already listed
Also added Gareth's particle workshop mod
Was putting this off for a while and finally got the motivation to do this 



Was unsure of how to handle the Discord.Net dlls for [HeadlessTweaks](https://github.com/New-Project-Final-Final-WIP/HeadlessTweaks) so I put them in the artifacts array of the releases

EIA485 commented:
hey, for headless tweaks add all the discord.net dlls as dependencies like in #82
i should review this as to take some load off lj but want to be fully transparent. i am not affiliated with any of these mods development but did make the old pr for headless tweaks.

badhaloninja commented:
There we go, I've separated the Discord.Net into their own entries
And the binaries on my releases tab are from the [nuget](https://www.nuget.org/packages/Discord.Net.Webhook/) for Discord.Net automatically downloaded via visual studio
I think you can manually download them from [nuget.info](https://nuget.info/packages/Discord.Net.Webhook/3.7.2) but I wasn't sure how or if I should directly link to those files there

ljoonal commented on manifest.json:
```diff
@@ -3157,6 +3386,614 @@
+                }
+            }
+        },
+        "me.Gareth48.ParticleWorkshop": {
+            "name": "ParticleWorkshop",
+            "description": "Adds a wizard for creating particles with color and alpha over time.\nCreated by Gareth48 with some huge help from badhaloninja and Cyro!",
+            "category": "Wizards",
+            "sourceLocation": "https://github.com/Epimonster/ParticleWorkshop",
+            "authors": {
+                "Gareth48": {
+                    "url": "https://github.com/Epimonster"
+                },
+                "Cyro": {
+                    "url": "https://github.com/RileyGuy"
+                },
+                "badhaloninja": {
+                    "url": "https://github.com/badhaloninja"
+                }
+            },
+            "versions": {
+                "1.0.0": {
+                    "releaseUrl": "https://github.com/Epimonster/ParticleWorkshop/releases/tag/1.0.0",
+                    "artifacts": [
+                        {
+                            "url": "https://github.com/Epimonster/ParticleWorkshop/releases/download/1.0.0/ParticleWorkshop.dll",
+                            "filename": "ParticleWorkshop.dll",
+                            "sha256": "2e3ef8e5aefc9fb1e097f66437224b7af7ca8ac931a92a80342d8175df14a32c"
+                        }
+                    ] 
+                }
+            }
+        },
```
👍

ljoonal commented on manifest.json:
```diff
@@ -3157,6 +3386,614 @@
+                }
+            },
+            "versions": {
+                "1.2.0": {
```
Didn't review properly, but I think we can just trust it enough to not be malicious, since didn't see anything clearly out of place.

ljoonal commented on manifest.json:
```diff
@@ -3157,6 +3386,614 @@
+                }
+            }
+        },
+        "me.badhaloninja.CreateNewAdditions": {
+            "name": "CreateNewAdditions",
+            "description": "Adds more primitives to the CreateNewDialog",
+            "category": "Technical Tweaks",
+            "sourceLocation": "https://github.com/badhaloninja/CreateNewAdditions",
+            "authors": {
+                "badhaloninja": {
+                    "url": "https://github.com/badhaloninja"
+                }
+            },
+            "versions": {
+                "1.0.0": {
+                    "releaseUrl": "https://github.com/badhaloninja/CreateNewAdditions/releases/tag/v1.0.0",
+                    "artifacts": [
+                        {
+                            "url": "https://github.com/badhaloninja/CreateNewAdditions/releases/download/v1.0.0/CreateNewAdditions.dll",
+                            "filename": "CreateNewAdditions.dll",
+                            "sha256": "9de58b2afb452f7f881b8035c0abc80c5ab676f0a34b2ee3f060b55b3d47a5b4"
+                        }
+                    ] 
+                }
+            }
+        },
```
👍

ljoonal commented on manifest.json:
```diff
@@ -3157,6 +3386,614 @@
+                }
+            }
+        },
+        "me.badhaloninja.InspectorExpandAll": {
+            "name": "InspectorExpandAll",
+            "description": "Adds a button to the inspector to expand all visible slots on an inspector",
+            "category": "Inspectors",
+            "sourceLocation": "https://github.com/badhaloninja/InspectorExpandAll",
+            "authors": {
+                "badhaloninja": {
+                    "url": "https://github.com/badhaloninja"
+                }
+            },
+            "versions": {
+                "1.0.0": {
+                    "releaseUrl": "https://github.com/badhaloninja/InspectorExpandAll/releases/tag/v1.0.0",
+                    "artifacts": [
+                        {
+                            "url": "https://github.com/badhaloninja/InspectorExpandAll/releases/download/v1.0.0/InspectorExpandAll.dll",
+                            "filename": "InspectorExpandAll.dll",
+                            "sha256": "a8deefcefb252abcf2baac7046293cb9b30aae533c66c8de4166af8a58ce61ba"
+                        }
+                    ] 
+                }
+            }
+        },
```
👍

ljoonal commented on manifest.json:
```diff
@@ -3157,6 +3386,614 @@
+                }
+            }
+        },
+        "me.badhaloninja.PimaxSwordAsViveWand": {
+            "name": "PimaxSwordAsViveWand",
+            "description": "Uses vive wand bindings for pimax sword controllers\nRequested by Engi",
+            "category": "Keybinds & Gestures",
+            "sourceLocation": "https://github.com/badhaloninja/PimaxSwordAsViveWand",
+            "authors": {
+                "badhaloninja": {
+                    "url": "https://github.com/badhaloninja"
+                }
+            },
+            "versions": {
+                "1.0.0": {
+                    "releaseUrl": "https://github.com/badhaloninja/PimaxSwordAsViveWand/releases/tag/v1.0.0",
+                    "artifacts": [
+                        {
+                            "url": "https://github.com/badhaloninja/PimaxSwordAsViveWand/releases/download/v1.0.0/PimaxSwordAsViveWand.dll",
+                            "filename": "PimaxSwordAsViveWand.dll",
+                            "sha256": "d1cc83dfd787cd004528164867499f44cab32ae7d48599f19617d7cc68449bb9"
+                        }
+                    ] 
+                }
+            }
+        },
```
👍

ljoonal commented on manifest.json:
```diff
@@ -3157,6 +3386,614 @@
+                }
+            }
+        },
+        "me.badhaloninja.ExitNeosAndShutdown": {
+            "name": "ExitNeosAndShutdown",
+            "description": "Adds a button to the exitscreen that shuts down your computer",
+            "category": "Memes",
+            "sourceLocation": "https://github.com/badhaloninja/ExitNeosAndShutdown",
+            "authors": {
+                "badhaloninja": {
+                    "url": "https://github.com/badhaloninja"
+                }
+            },
+            "versions": {
+                "1.0.0": {
+                    "releaseUrl": "https://github.com/badhaloninja/ExitNeosAndShutdown/releases/tag/v1.0.0",
+                    "artifacts": [
+                        {
+                            "url": "https://github.com/badhaloninja/ExitNeosAndShutdown/releases/download/v1.0.0/ExitNeosAndShutdown.dll",
+                            "filename": "ExitNeosAndShutdown.dll",
+                            "sha256": "478372db2e907a53029c531751e95bc002a99ac8ac3b3c753b664fc991c3d533"
+                        }
+                    ] 
+                }
+            }
+        },
```
Uses hardcoded neosdb urls & seems windows specific, so missing broken on linux flag, or making a OS check and adding `shutdown now` when running on linux

ljoonal commented on manifest.json:
```diff
@@ -3157,6 +3386,614 @@
+                }
+            }
+        },
+        "me.badhaloninja.ExtraTypeAliases": {
+            "name": "ExtraTypeAliases",
+            "description": "Adds more type aliases to the component attacher",
+            "category": "Technical Tweaks",
+            "sourceLocation": "https://github.com/badhaloninja/ExtraTypeAliases",
+            "authors": {
+                "badhaloninja": {
+                    "url": "https://github.com/badhaloninja"
+                }
+            },
+            "versions": {
+                "1.0.0": {
+                    "releaseUrl": "https://github.com/badhaloninja/ExtraTypeAliases/releases/tag/v1.0.0",
+                    "artifacts": [
+                        {
+                            "url": "https://github.com/badhaloninja/ExtraTypeAliases/releases/download/v1.0.0/ExtraTypeAliases.dll",
+                            "filename": "ExtraTypeAliases.dll",
+                            "sha256": "ead86d2c05d8366b97aa434064601685fe28c5524ef7c3edc3e4c0dac0192404"
+                        }
+                    ] 
+                }
+            }
+        },
```
👍

ljoonal commented on manifest.json:
```diff
@@ -3157,6 +3386,614 @@
+                        }
+                    ] 
+                },
+                "1.2.0": {
+                    "changelog": "Added the option to auto scoop materials when created via the CreateNewDialog",
+                    "releaseUrl": "https://github.com/badhaloninja/MaterialTipAutoScoop/releases/tag/v1.2.0",
+                    "artifacts": [
+                        {
+                            "url": "https://github.com/badhaloninja/MaterialTipAutoScoop/releases/download/v1.2.0/MaterialTipAutoScoop.dll",
+                            "filename": "MaterialTipAutoScoop.dll",
+                            "sha256": "29c85787a2a3e6ab40d9f71b89103f7c5437b2e1aad9571c88ec3c5c6abc8918"
+                        }
+                    ] 
+                },
+                "1.1.0": {
+                    "changelog": "Config support",
+                    "releaseUrl": "https://github.com/badhaloninja/MaterialTipAutoScoop/releases/tag/v1.1.0",
+                    "artifacts": [
+                        {
+                            "url": "https://github.com/badhaloninja/MaterialTipAutoScoop/releases/download/v1.1.0/MaterialTipAutoScoop.dll",
+                            "filename": "MaterialTipAutoScoop.dll",
+                            "sha256": "f604b91d63d2f3d14419d772a74697ba2ddcc0a846709e201dff55b8147f08ce"
+                        }
+                    ] 
+                },
+                "1.0.0": {
+                    "releaseUrl": "https://github.com/badhaloninja/MaterialTipAutoScoop/releases/tag/v1.0.0",
+                    "artifacts": [
+                        {
+                            "url": "https://github.com/badhaloninja/MaterialTipAutoScoop/releases/download/v1.0.0/MaterialTipAutoScoop.dll",
+                            "filename": "MaterialTipAutoScoop.dll",
+                            "sha256": "d75dd288d04340f4c22d65ac2ada51def260d5c258ffb8ba6ec402a16a3cdbbf"
+                        }
+                    ] 
+                }
+            }
```
Didn't review old versions

ljoonal commented on manifest.json:
```diff
@@ -3157,6 +3386,614 @@
+                }
+            },
+            "versions": {
+                "1.2.1": {
```
👍 but Hardcodes neosdb url

ljoonal commented on manifest.json:
```diff
@@ -3157,6 +3386,614 @@
+                        }
+                    ] 
+                },
+                "1.0.0": {
+                    "releaseUrl": "https://github.com/badhaloninja/OculusSwapMenuAndSecondary/releases/tag/v1.0.0",
+                    "artifacts": [
+                        {
+                            "url": "https://github.com/badhaloninja/OculusSwapMenuAndSecondary/releases/download/v1.0.0/OculusSwapMenuAndSecondary.dll",
+                            "filename": "OculusSwapMenuAndSecondary.dll",
+                            "sha256": "2eda99e2de892fc80f33be903cda9d45ce2bf24ec91d99e65f2bed62ac9a567a"
+                        }
+                    ] 
+                }
```
Didn't review old version

ljoonal commented on manifest.json:
```diff
@@ -3157,6 +3386,614 @@
+                }
+            },
+            "versions": {
+                "1.1.0": {
+                    "changelog": "Rebind jump to ButtonXA",
+                    "releaseUrl": "https://github.com/badhaloninja/OculusSwapMenuAndSecondary/releases/tag/v1.1.0",
+                    "artifacts": [
+                        {
+                            "url": "https://github.com/badhaloninja/OculusSwapMenuAndSecondary/releases/download/v1.1.0/OculusSwapMenuAndSecondary.dll",
+                            "filename": "OculusSwapMenuAndSecondary.dll",
+                            "sha256": "8c1bafa0722dd4d36fc299c668a881b56185c6c4be5abb3162044ec1f7b0bee4"
+                        }
+                    ] 
+                },
```
👍

ljoonal commented on manifest.json:
```diff
@@ -3157,6 +3386,614 @@
+                }
+            }
+        },
+        "me.badhaloninja.NewMissingTexture": {
+            "name": "NewMissingTexture",
+            "description": "Changes the null texture",
+            "category": "Memes",
+            "sourceLocation": "https://github.com/badhaloninja/NewMissingTexture",
+            "authors": {
+                "badhaloninja": {
+                    "url": "https://github.com/badhaloninja"
+                }
+            },
+            "versions": {
+                "1.0.0": {
+                    "releaseUrl": "https://github.com/badhaloninja/NewMissingTexture/releases/tag/v1.0.0",
+                    "artifacts": [
+                        {
+                            "url": "https://github.com/badhaloninja/NewMissingTexture/releases/download/v1.0.0/NewMissingTexture.dll",
+                            "filename": "NewMissingTexture.dll",
+                            "sha256": "c298190a0f1033346f31d16959b216640f7c2000555e2cd5448af25a8ca4866a"
+                        }
+                    ] 
+                }
+            }
+        },
```
👍

ljoonal commented on manifest.json:
```diff
@@ -3157,6 +3386,614 @@
+                }
+            }
+        },
+        "me.badhaloninja.ReloadInspectorButton": {
+            "name": "ReloadInspectorButton",
+            "description": "Adds a button that reloads an inspector component view as local user",
+            "category": "Inspectors",
+            "sourceLocation": "https://github.com/badhaloninja/ReloadInspectorButton",
+            "authors": {
+                "badhaloninja": {
+                    "url": "https://github.com/badhaloninja"
+                }
+            },
+            "versions": {
+                "1.0.0": {
+                    "releaseUrl": "https://github.com/badhaloninja/ReloadInspectorButton/releases/tag/v1.0.0",
+                    "artifacts": [
+                        {
+                            "url": "https://github.com/badhaloninja/ReloadInspectorButton/releases/download/v1.0.0/ReloadInspectorButton.dll",
+                            "filename": "ReloadInspectorButton.dll",
+                            "sha256": "cea46202c8ba5180c542ef07635e4bc2d60c577f149ebb830f04444bb0382948"
+                        }
+                    ] 
+                }
+            }
+        },
```
👍

ljoonal commented on manifest.json:
```diff
@@ -3157,6 +3386,614 @@
+                }
+            },
+            "versions": {
+                "1.0.0": {
+                    "releaseUrl": "https://github.com/badhaloninja/DevOriginSelectTweaks/releases/tag/v1.0.0",
+                    "artifacts": [
+                        {
+                            "url": "https://github.com/badhaloninja/DevOriginSelectTweaks/releases/download/v1.0.0/DevOriginSelectTweaks.dll",
+                            "filename": "DevOriginSelectTweaks.dll",
+                            "sha256": "6aa068a5c4bfdfebdfbab0a81feacc75256608cfc5a80edafc91c3a4a356cb8e"
+                        }
+                    ] 
+                }
```
👍 but hardcoded neosdb url

ljoonal commented on manifest.json:
```diff
@@ -3157,6 +3386,614 @@
+                        }
+                    ] 
+                },
+                "1.0.0": {
+                    "releaseUrl": "https://github.com/badhaloninja/DeselectOwnGizmos/releases/tag/v1.0.0",
+                    "artifacts": [
+                        {
+                            "url": "https://github.com/badhaloninja/DeselectOwnGizmos/releases/download/v1.0.0/DeselectOwnGizmos.dll",
+                            "filename": "DeselectOwnGizmos.dll",
+                            "sha256": "00017e06bc2231f1cc1e3d10ef5b8c8ee27705a22fee5f101a7c1fc22b5cc78c"
+                        }
+                    ] 
+                }
```
Didn't review old version

ljoonal commented on manifest.json:
```diff
@@ -3157,6 +3386,614 @@
+                }
+            },
+            "versions": {
+                "1.1.0": {
+                    "changelog": " - Added new icon\n - Removed unnecessary traverse",
+                    "releaseUrl": "https://github.com/badhaloninja/DeselectOwnGizmos/releases/tag/v1.1.0",
+                    "artifacts": [
+                        {
+                            "url": "https://github.com/badhaloninja/DeselectOwnGizmos/releases/download/v1.1.0/DeselectOwnGizmos.dll",
+                            "filename": "DeselectOwnGizmos.dll",
+                            "sha256": "b58500c92be0401df3c406a88663ba0333581ffa009890dfbca550558e2bf43b"
+                        }
+                    ] 
+                },
```
👍 but hardcoded neosdb url

ljoonal commented on manifest.json:
```diff
@@ -3157,6 +3386,614 @@
+                }
+            }
+        },
+        "me.badhaloninja.BetterLUTImport": {
+            "name": "BetterLUTImport",
+            "description": "Make LUTs import as a LUT material",
+            "category": "Asset Importing Tweaks",
+            "sourceLocation": "https://github.com/badhaloninja/BetterLUTImport",
+            "authors": {
+                "badhaloninja": {
+                    "url": "https://github.com/badhaloninja"
+                }
+            },
+            "versions": {
+                "1.0.0": {
+                    "releaseUrl": "https://github.com/badhaloninja/BetterLUTImport/releases/tag/v1.0.0",
+                    "artifacts": [
+                        {
+                            "url": "https://github.com/badhaloninja/BetterLUTImport/releases/download/v1.0.0/BetterLUTImport.dll",
+                            "filename": "BetterLUTImport.dll",
+                            "sha256": "d7ec5da2311c16df0bc25a50cbbcee3e9f3ad626ad21b8ff3823b6a52d7d9b52"
+                        }
+                    ] 
+                }
+            }
+        },
```
👍

ljoonal commented on manifest.json:
```diff
@@ -3157,6 +3386,614 @@
+                        }
+                    ]
+                },
+                "1.0.1": {
+                    "changelog": "Removed dev log message",
+                    "releaseUrl": "https://github.com/badhaloninja/DesktopNotifications/releases/tag/v1.0.1",
+                    "artifacts": [
+                        {
+                            "url": "https://github.com/badhaloninja/DesktopNotifications/releases/download/v1.0.1/DesktopNotifications.dll",
+                            "filename": "DesktopNotifications.dll",
+                            "sha256": "fe56d7bbd5fa820469bf233952523071c1f5aa6effcbbf418ac64f287d3d5801"
+                        }
+                    ]
+                },
+                "1.0.0": {
+                    "releaseUrl": "https://github.com/badhaloninja/DesktopNotifications/releases/tag/v1.0.0",
+                    "artifacts": [
+                        {
+                            "url": "https://github.com/badhaloninja/DesktopNotifications/releases/download/v1.0.0/DesktopNotifications.dll",
+                            "filename": "DesktopNotifications.dll",
+                            "sha256": "89a3f251ea53efdd6207769076d2157f053ca6a64ff6b132d2c735c391aaec15"
+                        }
+                    ]
+                }
```
Didn't review old versions

ljoonal commented on manifest.json:
```diff
@@ -3157,6 +3386,614 @@
+                }
+            },
+            "versions": {
+                "1.1.0": {
+                    "changelog": "Switch to using float2 instead of 2 separate float options",
+                    "releaseUrl": "https://github.com/badhaloninja/DesktopNotifications/releases/tag/v1.1.0",
+                    "artifacts": [
+                        {
+                            "url": "https://github.com/badhaloninja/DesktopNotifications/releases/download/v1.1.0/DesktopNotifications.dll",
+                            "filename": "DesktopNotifications.dll",
+                            "sha256": "547426ba0ff8f02a15bd63031838d2ace58b5c65e5b0085096e0a09df20e9ce6"
+                        }
+                    ]
+                },
```
IL transpiler stuff though looks fine as it only ever removes a code instruction.

ljoonal commented on manifest.json:
```diff
@@ -95,6 +95,123 @@
+                "file"
+            ],
+            "versions": {
+                "3.7.2.0": {
+                    "releaseUrl": "https://github.com/discord-net/Discord.Net/releases/tag/3.7.2",
+                    "artifacts": [
+                        {
+                            "url": "https://github.com/New-Project-Final-Final-WIP/HeadlessTweaks/releases/download/v1.2.0/Discord.Net.Rest.dll",
+                            "filename": "Discord.Net.Rest.dll",
+                            "sha256": "ff9d22adbe079ca3594adc1d32149eb1fbe42656a8cccb68316672b60c94e399",
+                            "installLocation": "/nml_libs"
+                        }
+                    ]
+                },
```
Sha256 mashes upstream, so 👍 even though didn't review properly

ljoonal commented on manifest.json:
```diff
@@ -95,6 +95,123 @@
+                "file"
+            ],
+            "versions": {
+                "3.7.2.0": {
+                    "releaseUrl": "https://github.com/discord-net/Discord.Net/releases/tag/3.7.2",
+                    "artifacts": [
+                        {
+                            "url": "https://github.com/New-Project-Final-Final-WIP/HeadlessTweaks/releases/download/v1.2.0/Discord.Net.Core.dll",
+                            "filename": "Discord.Net.Core.dll",
+                            "sha256": "68cf9f25959b675a213d642fe2d6376bb057cc25c4d3e4093409761afc772e69",
+                            "installLocation": "/nml_libs"
+                        }
+                    ]
+                },
```
Sha256 matches upstream so 👍

ljoonal reviewed (changes requested):
If it'd be acceptable, could we only add the artifacts only for the latest versions? As reviewing old versions adds extra work. I don't ever see a reason to add the artifacts for anything but the latest working version.

badhaloninja commented:
Alright, I've gotten rid of all of the non reviewed old versions
Also added linux broken flags to ExitNeosAndShutDown
And fixed the filename for Discord.Net.Webhook

ljoonal commented:
> parse error: Expected separator between values at line 3533, column 43
